### PR TITLE
Fix #7945: Add cost of clearing the sloped tile when placing a dock

### LINKED ISSUE #7945: Building docks does not account the cost of clearing the sloped tile
## Version of OpenTTD
1.10.0-beta2
## Expected result
Cost of dock + cost of clearing sloped tile
## Actual result
Cost of dock
## Steps to reproduce
Dock 1: Build a dock, see total cost
Dock 2: Clear sloped tile before building a dock, see total cost

### PR DESCRIPTION
**Pull request by SamuXarick**
#7945

### CODE REVIEW

**Comment by LordAro** [comment]
I think this should be 2 commits - the first 2 - (Add cost of... + regression fix) & and the next 2 (Using "Change: Allow..." commit message). Can you squash together accordingly?

**Review by LordAro — DISMISSED** [review]
I think this is fine, but I'd like to see a review from someone else first...

**Review by Yexo — DISMISSED** [review]

File: src/station_cmd.cpp [hunk 1]
```diff
@@ -2538,12 +2538,14 @@ CommandCost CmdBuildDock(TileIndex tile, DoCommandFlag flags, uint32 p1, uint32
...
 
+	CommandCost cost(EXPENSES_CONSTRUCTION, _price[PR_BUILD_STATION_DOCK]);
 	ret = DoCommand(tile, 0, 0, flags, CMD_LANDSCAPE_CLEAR);
 	if (ret.Failed()) return ret;
+	cost.AddCost(ret);
 
 	TileIndex tile_cur = tile + TileOffsByDiagDir(direction);
 
-	if (!IsTileType(tile_cur, MP_WATER) || !IsTileFlat(tile_cur)) {
+	if (!HasTileWaterGround(tile_cur) || !IsTileFlat(tile_cur)) {
```
Comment (glx22) [inline src/station_cmd.cpp]: This change is wrong, `cur_tile` must be a `MP_WATER` tile, it's the dock end.

**Review by glx22 — DISMISSED** [review]

**Comment by nielsmh** [comment]
Needs conflict resolved in regression test `result.txt` before this can merge. (The file was moved in the CMake change.)

**Comment by TrueBrain** [comment]
I removed the second commit from this PR; I have no clue what it is doing, why it is there, or why we would want it. It contains no documentation of the change, no reference to what it fixes, etc etc etc. Seems some people never learn ;)

The other commit is valid, and resolves a bug.

**Review by TrueBrain — APPROVED** [review]

**Pull request merged**